src/common/sentry.ts

```diff
@@ -20,73 +20,75 @@
 }
 
 /**
  * Initialize Sentry for the current process so to send errors for debugging.
  */
-export async function SentryInit(): Promise<void> {
+export function SentryInit() {
   try {
-    const Sentry = await requireSentry();
+    const processName = mapProcessName(process.type);
 
-    try {
-      Sentry.init({
-        beforeSend: event => {
-          if (UserStore.getInstance().allowErrorReporting) {
-            return event;
-          }
+    Sentry.init({
+      beforeSend: (event) => {
+        // default to false, in case instance of UserStore is not created (yet)
+        let allowErrorReporting = false;
 
-          logger.info(`🔒  [SENTRY-BEFORE-SEND-HOOK]: allowErrorReporting: false. Sentry event is caught but not sent to server.`);
-          logger.info("🔒  [SENTRY-BEFORE-SEND-HOOK]: === START OF SENTRY EVENT ===");
-          logger.info(event);
-          logger.info("🔒  [SENTRY-BEFORE-SEND-HOOK]: ===  END OF SENTRY EVENT  ===");
+        try {
+          allowErrorReporting = UserStore.getInstance()?.allowErrorReporting;
+        } catch {
+          // ignore the TypeError throw by UserStore.getInstance()
+        }
 
-          // if return null, the event won't be sent
-          // ref https://github.com/getsentry/sentry-javascript/issues/2039
-          return null;
-        },
-        dsn: sentryDsn,
-        integrations: [
-          new CaptureConsole({ levels: ["error"] }),
-          new Dedupe(),
-          new Offline()
-        ],
-        initialScope: {
-          tags: {
-            // "translate" browser to 'main' as Lens developer more familiar with the term 'main'
-            "process": process.type === "browser" ? "main" : "renderer"
-          }
-        },
-        environment: isProduction ? "production" : "development",
-      });
+        if (allowErrorReporting) {
+          return event;
+        }
 
-      sentryIsInitialized = true;
+        logger.info(`🔒  [SENTRY-BEFORE-SEND-HOOK]: allowErrorReporting: ${allowErrorReporting}. Sentry event is caught but not sent to server.`);
+        logger.info("🔒  [SENTRY-BEFORE-SEND-HOOK]: === START OF SENTRY EVENT ===");
+        logger.info(event);
+        logger.info("🔒  [SENTRY-BEFORE-SEND-HOOK]: ===  END OF SENTRY EVENT  ===");
 
-      logger.info(`✔️  [SENTRY-INIT]: Sentry for ${process.type} is initialized.`);
-    } catch (error) {
-      logger.warn(`⚠️  [SENTRY-INIT]: Sentry.init() error: ${error?.message ?? error}.`);
+        // if return null, the event won't be sent
+        // ref https://github.com/getsentry/sentry-javascript/issues/2039
+        return null;
+      },
+      dsn: sentryDsn,
+      integrations: [
+        new CaptureConsole({ levels: ["error"] }),
+        new Dedupe(),
+        new Offline()
+      ],
+      initialScope: {
+        tags: {
+
+          "process": processName
+        }
+      },
+      environment: isProduction ? "production" : "development",
+    });
+
+    if (sentryDsnIsValid) {
```

Comment:
If this is only needed for the `""` case can't we just check for that?

Reply:
Yes.. but I guess regex covers more cases and more future proof?

Reply:
But from looking at the sentry code they already run this regex so how is it more future proof?

Reply:
For example, Sentry changes their source code in 7.0 and don't throw error from Sentry.init() when dsn is not valid, we will have side effects for showing the UI even if dsn is not valid after upgrading to 7.0.

Reply:
What if Sentry changes its format of the dsn in v7? I don't find the argument in a future version they may no longer throw an error persuasive. I actually find the fact that they don't throw an error on `""` strange.

Reply:
Ideally they export the regex check for dsn somewhere, but at least I can't find it for now. I don't know why they didn't throw "". This happened to be our case.

Reply:
They do report an error to console, I am seeing the following on master (with `sentryDsn: ""` in package.json):

```
SentryError: Attempted to enable Electron native crash reporter but no DSN was supplied
[0]     at new SentryError (/repos/lens/node_modules/@sentry/utils/dist/error.js:9:28)
[0]     at MainBackend.<anonymous> (/repos/lens/node_modules/@sentry/electron/dist/main/backend.js:152:27)
[0]     at step (/repos/lens/node_modules/@sentry/electron/node_modules/tslib/tslib.js:143:27)
[0]     at Object.next (/repos/lens/node_modules/@sentry/electron/node_modules/tslib/tslib.js:124:57)
[0]     at /repos/lens/node_modules/@sentry/electron/node_modules/tslib/tslib.js:117:75
[0]     at new Promise (<anonymous>)
[0]     at Object.__awaiter (/repos/lens/node_modules/@sentry/electron/node_modules/tslib/tslib.js:113:16)
[0]     at MainBackend._installNativeHandler (/repos/lens/node_modules/@sentry/electron/dist/main/backend.js:146:24)
[0]     at new MainBackend (/repos/lens/node_modules/@sentry/electron/dist/main/backend.js:66:34)
[0]     at MainClient.BaseClient (/repos/lens/node_modules/@sentry/core/dist/baseclient.js:52:25) {
[0]   name: 'SentryError'
[0] }
```

Reply:
Yes, but it didn't throw error from Sentry.init(), at least the try...catch can't catch it

Reply:
It is this line https://github.com/getsentry/sentry-electron/blob/edd2775048d4658f88fd23a989574da913e6762a/src/main/backend.ts#L101

Reply:
I think this is a bug, it should reject the init

Reply:
Forget = Consumes the promise and logs the error when it rejects.

https://github.com/getsentry/sentry-javascript/blob/5ed31354/packages/utils/src/async.ts#L6

Nice name...

Reply:
I am not sure if it's a bug, if DSN can be set dynamically at runtime, there is no need to reject promise if the initial dsn is not valid.

Reply:
Yeah... a really bad coding practice.

Reply:
Submitted https://github.com/getsentry/sentry-electron/issues/348

Reply:
How would that work, calling `init` again? I cannot find anything in the docs to state that you can pass in a dsn without calling `init`

Reply:
Yeah, then it's probably a bug, or bad-by-design

Reply:
should be easy to fix too.

Reply:
Submitted fix https://github.com/getsentry/sentry-electron/pull/349

Reply:
So what's the plan, we should wait for Sentry to merge your PR and update Sentry libs and remove the "" check in this PR?

Reply:
I think that would take to long. How about keeping just the `""` check (since we do get errors thrown on invalid URLs of length > 0) and reverting to assigning to a `sentryIsInitialized` variable. 

All the while adding a `// TODO:` that references that PR.

Reply:
WDYT?

Reply:
I am thinking just remove the Sentry init check, and always shows the UI (checkbox) in preference page, for users, it doesn't matter Sentry has initiated or not.

I imagine users only care if they can disable exceptions transporting.

Reply:
Yes probably, and if it isn't initialized then even if it is true it won't send anything.

Reply:
> I am thinking just remove the Sentry init check, and always shows the UI (checkbox) in preference page, for users, it doesn't matter Sentry has initiated or not.
> 
> I imagine users only care if they can disable exceptions transporting.

I think we should keep UI hidden if `sentryDsn` is not set (or is empty).

Reply:
> > I am thinking just remove the Sentry init check, and always shows the UI (checkbox) in preference page, for users, it doesn't matter Sentry has initiated or not.
> > I imagine users only care if they can disable exceptions transporting.
> 
> I think we should keep UI hidden if `sentryDsn` is not set (or is empty).

Ok, let me summerise a bit...

if we want:

> Keep UI hidden if `sentryDsn` is not set (or is empty).

We have two ways

1. Since `Sentry.init` won't throw exception if DSN is not set (or is empty). Keep `sentryDsnIsValid` and use that to hide UI if false. Remove `sentryIsInitialized` as it will always be true because `Sentry.init()` never throws.

2. Wait for Sentry to merge their PR https://github.com/getsentry/sentry-electron/issues/348, and update Sentry to the version with "throw exception" behaviour, remove `sentryDsnIsValid` as not needed, keep `sentryIsInitialized` and use `sentryIsInitialized` to hide UI if false.

Reply:
https://github.com/getsentry/sentry-electron/pulls has only 2 PRs opened by this time, so big chance that the issue will be solved quickly. I'd prefer second variant if it will be resolved within 1-2 d. Make option 1 otherwise (and keep and eye on PR)

Reply:
a third options, only hide preference UI if sentryDsn is an empty string or undefined/null, remove try...catch for `Sentry.init()` (should be caught by tests instead).  https://github.com/lensapp/lens/pull/3325/commits/ec8c3bc26d2a5ebe0622122bbbec2a3a215bf302

Reply:
Might be the best one. Is think its rare case to have invalid dsn value whatsoever.